srtcore/api.cpp

```diff
@@ -233,15 +228,24 @@ string srt::CUDTUnited::CONID(SRTSOCKET sock)
     return os.str();
 }
 
-int srt::CUDTUnited::startup()
+void srt::CUDTUnited::implicitStartup()
+{
+    startup(true);
+}
+
+int srt::CUDTUnited::startup(bool implicit)
 {
     ScopedLock gcinit(m_InitLock);
-    if (m_bGCStatus)
+
+    if (implicit && m_iInstanceCount > 0)
```

Comment:
I would add a comment to explain explicitly what happened here.

```suggestion

    // For implicit startup (from srt_create_socket/group) never increase instance count
    if (implicit && m_iInstanceCount > 0)
```